homework02/test/examples/Main.hs

```diff
@@ -1,15 +1,39 @@
-module Main
-  ( main
-  )
-  where
-
--- doctest
-import qualified Test.DocTest as DocTest
+module Main where
 
+import Log
+import LogAnalysis
+import Test.Hspec
 
 main :: IO ()
-main =
-  DocTest.doctest
-    [ "-isrc"
-    , "src/LogAnalysis.hs"
-    ]
+main = hspec $ do
+
+  describe "parseMessage" $ do
+    it "parses an error message" $
+      parseMessage "E 2 562 help help" `shouldBe` LogMessage (Error 2) 562 "help help"
+
+    it "parses an info message" $
+      parseMessage "I 29 la la la" `shouldBe` LogMessage Info 29 "la la la"
+
+    it "parses an Unknown message" $
+      parseMessage "This is not in the right format" `shouldBe` Unknown "This is not in the right format"  
+
+  describe "insert" $
+    it "returns a tree with LogMessage inserted" $ do
+      let tree = Node Leaf (LogMessage (Error 2) 562 "help help") Leaf
+      insert (LogMessage Info 29 "la") tree `shouldBe` (Node (Node Leaf (LogMessage Info 29 "la") Leaf) (LogMessage (Error 2) 562 "help help") Leaf)
+   
+  describe "build" $ 
+    it "builds up a MessageTree containing the messages in a list" $ do
+      let messageTreeSet = [LogMessage (Error 2) 562 "help help", LogMessage Info 29 "la"]
+      build messageTreeSet `shouldBe` Node (Node Leaf (LogMessage Info 29 "la") Leaf) (LogMessage (Error 2) 562 "help help") Leaf
```

Comment:
the same as above, please split this into multiple lines to make it more readable